tests/test_oneclick.py

```diff
@@ -0,0 +1,295 @@
+
+        ids = [(x.type, x.identifier) for x in metadata.identifiers]
+
+        # The original data contains a blank ASIN in addition to the
+        # actual ASIN, but it doesn't show up here.
+        eq_(
+            [
+                (Identifier.ISBN, "9780307378101"),
+            ],
+            sorted(ids)
+        )
+
+        # Available formats.      
+        [epub] = sorted(metadata.circulation.formats, key=lambda x: x.content_type)        
+        eq_(Representation.EPUB_MEDIA_TYPE, epub.content_type)       
+        eq_(DeliveryMechanism.ONECLICK_DRM, epub.drm_scheme)      
+
+        # Links to various resources.
+        shortd, image = sorted(
+            metadata.links, key=lambda x:x.rel
+        )
+
+        eq_(Hyperlink.SHORT_DESCRIPTION, shortd.rel)
+        assert shortd.content.startswith("THE NO. 1 LADIES' DETECTIVE AGENCY")
+
+        eq_(Hyperlink.IMAGE, image.rel)
+        eq_('http://images.oneclickdigital.com/EB00148140/EB00148140_image_128x192.jpg', image.href)
+
+        thumbnail = image.thumbnail
+
+        eq_(Hyperlink.THUMBNAIL_IMAGE, thumbnail.rel)
+        eq_('http://images.oneclickdigital.com/EB00148140/EB00148140_image_95x140.jpg', thumbnail.href)
+
+        # Note: For now, no measurements associated with the book.
+
+        # Request only the bibliographic information.
+        metadata = OneClickRepresentationExtractor.isbn_info_to_metadata(datadict, include_bibliographic=True, include_formats=False)
+        eq_("Tea Time for the Traditionally Built", metadata.title)
+        eq_(None, metadata.circulation)
+
+        # Request only the format information.
+        metadata = OneClickRepresentationExtractor.isbn_info_to_metadata(datadict, include_bibliographic=False, include_formats=True)
+        eq_(None, metadata.title)
+        [epub] = sorted(metadata.circulation.formats, key=lambda x: x.content_type)        
+        eq_(Representation.EPUB_MEDIA_TYPE, epub.content_type)       
+        eq_(DeliveryMechanism.ONECLICK_DRM, epub.drm_scheme)      
+
+
+
+class TestOneClickBibliographicCoverageProvider(OneClickTest):
+    """Test the code that looks up bibliographic information from OneClick."""
+
+    def setup(self):
+        super(TestOneClickBibliographicCoverageProvider, self).setup()
+
+        self.provider = OneClickBibliographicCoverageProvider(
+            self._db, oneclick_api=self.api
+        )
+
+
+    def test_script_instantiation(self):
+        # Test that RunCoverageProviderScript can instantiate
+        # the coverage provider.
+        
+        script = RunCoverageProviderScript(
+            OneClickBibliographicCoverageProvider, self._db, [],
+            oneclick_api=self.api
+        )
+        assert isinstance(script.provider, 
+                          OneClickBibliographicCoverageProvider)
+        eq_(script.provider.api, self.api)
+
+
+    def test_invalid_or_unrecognized_guid(self):
+        # A bad or malformed GUID can't get coverage.
```

Comment:
There's some leftover code from Overdrive here. Can you change these error messages to the messages you see from One Click when you send an invalid ISBN?

Also change "guid" to "ISBN".